optimum/commands/export/openvino.py

```diff
@@ -68,8 +68,21 @@ def parse_args_openvino(parser: "ArgumentParser"):
             "This is needed by some models, for some tasks. If not provided, will attempt to use the tokenizer to guess it."
         ),
     )
-    optional_group.add_argument("--fp16", action="store_true", help="Compress weights to fp16"),
-    optional_group.add_argument("--int8", action="store_true", help="Compress weights to int8"),
+    optional_group.add_argument(
+        "--weight-format",
+        type=str,
+        choices=["f32", "f16", "i8", "i4_sym_g128", "i4_asym_g128", "i4_sym_g64", "i4_asym_g64"],
```

Comment:
also we might want to move the sym/asym from this set of options so that it can also be made available for int8, not sure it's needed though the default asym mode might be enough, let me know what you think

Reply:
We will have INT8 symmetric in the new version of NNCF. I am also thinking that we need to reduce the number of available options here and keep only symmetrical because they provide a better accuracy-performance trade-off (varying group size and ratio). @ljaljushkin, please provide your opinion as well.

Reply:
symmetric mode has a better correlation between model size and latency than asymmetric one.
can't say for sure, that varying group size and ratio for symmetric always gives a decent accuracy-performance trade-off.
there are some models when symmetric mode doesn't achieve it.
![image](https://github.com/huggingface/optimum-intel/assets/4014476/46dce16c-b2ef-4d28-9f4c-55ebe8f5250d)

Reply:
> Would be in favor of using fp32, fp16, int8 to keep the same format as for [transformers](https://github.com/huggingface/transformers/blob/v4.36.1/src/transformers/training_args.py#L900) and [optimum](https://github.com/huggingface/optimum/blob/7376c6ad7f0a852cb81dfbb9e9539eb6ed14fd36/optimum/exporters/tflite/base.py#L52)

I followed the notation of OpenVINO [types](https://github.com/huggingface/optimum-intel/blob/198149983fb6db2f3cd8bd616fa0189c2c6d9748/optimum/intel/openvino/utils.py#L47) but I can change.

Reply:
> I am also thinking that we need to reduce the number of available options here
I would much prefer to be able to use all weight compression options available in NNCF in Optimum. In my experience there are always specific cases where they are useful, and it's not good to have to completely switch frameworks/APIs when you want to use them. Also agreed that we should not overwhelm users - but in my opinion we're not there yet - and it also introduces confusion if there are differences between what's available in NNCF and what's available in optimum.

Reply:
> > Would be in favor of using fp32, fp16, int8 to keep the same format as for [transformers](https://github.com/huggingface/transformers/blob/v4.36.1/src/transformers/training_args.py#L900) and [optimum](https://github.com/huggingface/optimum/blob/7376c6ad7f0a852cb81dfbb9e9539eb6ed14fd36/optimum/exporters/tflite/base.py#L52)
> 
> I followed the notation of OpenVINO [types](https://github.com/huggingface/optimum-intel/blob/198149983fb6db2f3cd8bd616fa0189c2c6d9748/optimum/intel/openvino/utils.py#L47) but I can change.

I think that would be easier to keep consistency with other optimum's subpackage. 

No strong opinion concerning the symmetric/asymmetric mode, I'm fine with both options

Reply:
> > I am also thinking that we need to reduce the number of available options here
> 
> I would much prefer to be able to use all weight compression options available in NNCF in Optimum. In my experience there are always specific cases where they are useful, and it's not good to have to completely switch frameworks/APIs when you want to use them. Also agreed that we should not overwhelm users - but in my opinion we're not there yet - and it also introduces confusion if there are differences between what's available in NNCF and what's available in optimum.

Thanks, Helena! Understood your concerns but we have experimental schemes in NNCF that are not yet performant in OpenVINO so I am not going to expose them at this point.

Reply:
> > > Would be in favor of using fp32, fp16, int8 to keep the same format as for [transformers](https://github.com/huggingface/transformers/blob/v4.36.1/src/transformers/training_args.py#L900) and [optimum](https://github.com/huggingface/optimum/blob/7376c6ad7f0a852cb81dfbb9e9539eb6ed14fd36/optimum/exporters/tflite/base.py#L52)
> > 
> > 
> > I followed the notation of OpenVINO [types](https://github.com/huggingface/optimum-intel/blob/198149983fb6db2f3cd8bd616fa0189c2c6d9748/optimum/intel/openvino/utils.py#L47) but I can change.
> 
> I think that would be easier to keep consistency with other optimum's subpackage.
> 
> No strong opinion concerning the symmetric/asymmetric mode, I'm fine with both options

To strong objections, I can align the names of precisions with other parts of HF ecosystem.

Reply:
updated names